data_pipeline_api/java_api/src/test/java/uk/ramp/metadata/HashVerificationMetadataSelectorTest.java

```diff
@@ -0,0 +1,33 @@
+package uk.ramp.metadata;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.Before;
+import org.junit.Test;
+import uk.ramp.hash.Hasher;
+
+public class HashVerificationMetadataSelectorTest {
+  private MetadataSelector metadataSelector;
+  private Hasher hasher;
+
+  @Before
+  public void setUp() {
+    this.metadataSelector = mock(MetadataSelector.class);
+    this.hasher = mock(Hasher.class);
+  }
+
+  @Test
+  public void testOpenForRead() {
+    var query = ImmutableMetadataItem.builder().filename("file1").verifiedHash("hash1").build();
+
+    when(metadataSelector.find(query)).thenReturn(query);
+    when(hasher.hash("file1")).thenReturn("hash1");
+
+    var hashVerificationFileApi =
+        new HashVerificationMetadataSelector(metadataSelector, hasher, true);
+
+    assertThat(hashVerificationFileApi.find(query)).isEqualTo(query.withCalculatedHash("hash1"));
+  }
```

Comment:
Also, you're not actually testing anywhere that you're using SHA-1 hashes, which could be important, but I guess it would quickly become obvious if you got that wrong, as soon as anyone starts using the API.

Reply:
Thanks John, I'll add tests in for both your points mentioned.